wire-ios-request-strategy/Sources/Synchronization/Decoding/StoreUpdateEvent.swift

```diff
@@ -95,6 +112,27 @@ public final class StoredUpdateEvent: NSManagedObject {
         return storedEvent
     }
 
+    static private func storedEventExists(for updateEventHash: Int, in context: NSManagedObjectContext) -> Bool {
+
+        let storedEvents = StoredUpdateEvent.events(eventHash: updateEventHash, context: context)
+        guard !storedEvents.isEmpty else {
+            return false
+        }
+        
+        for storedEvent in storedEvents where storedEvent.eventHash == updateEventHash || storedEvent.eventHash == 0 {
```

Comment:
note that if the storedEvent has no hash (=0), we fallback to just match by event id, in case we just upgraded to this version and had storedEvents from previous fetch in the database). @johnxnguyen does that make sense to you?

Reply:
Yes it makes sense.

Reply:
Why do we have the `where` clause here?  Don't all events in `storedEvents` already have the same event hash?

Reply:
right, I forget about updating the fetch query, https://github.com/wireapp/wire-ios/pull/1725/commits/7c8a940c85a45b5d3d60d2b8b9cff3154ec681cc